panel/io/state.py

```diff
@@ -598,6 +598,33 @@ def add_periodic_callback(
             self._periodic[self.curdoc].append(cb)
         return cb
 
+    @contextmanager
+    def block_expiration(self):
+        """
+        Blocks expiration of the current session, if used as a context manager
+        it will be unblocked afterwards.
+        """
+        if self.curdoc is None or self.curdoc.session_context is None:
+            yield
+            return
+
+        for session in self.curdoc.session_context.server_context.sessions:
+            session.block_expiration()
+        try:
+            yield
+        finally:
+            for session in self.curdoc.session_context.server_context.sessions:
+                session.unblock_expiration()
+
+    def unblock_expiration(self):
+        """
+        Unblock the expiration of the current session.
+        """
+        if self.curdoc is None or self.curdoc.session_context is None:
+            return
+        for session in self.curdoc.session_context.server_context.sessions:
+            session.unblock_expiration()
```

Comment:
```suggestion
```

Is this needed anymore?

Reply:
Yes, you may not want to use block_expiration as a context manager.

Reply:
No you are correct. Will fix.